sinch/core/adapters/httpx_adapter.py

```diff
@@ -0,0 +1,60 @@
+import httpx
+from sinch.core.ports.http_transport import AsyncHTTPTransport, HttpRequest
+from sinch.core.endpoint import HTTPEndpoint
+from sinch.core.models.http_response import HTTPResponse
+
+
+class HTTPXTransport(AsyncHTTPTransport):
+    def __init__(self, sinch):
+        super().__init__(sinch)
+        self.http_session = None
+
+    async def request(self, endpoint: HTTPEndpoint) -> HTTPResponse:
+        request_data: HttpRequest = self.prepare_request(endpoint)
+        request_data: HttpRequest = await self.authenticate(endpoint, request_data)
+
+        if not self.http_session:
+            self.http_session = httpx.AsyncClient()
+
+        self.sinch.configuration.logger.debug(
+            f"Async HTTP {request_data.http_method} call with headers:"
+            f" {request_data.headers} and body: {request_data.request_body} to URL: {request_data.url}"
+        )
+
+        if isinstance(request_data.request_body, str):
+            response = await self.http_session.request(
+                method=request_data.http_method,
+                headers=request_data.headers,
+                url=request_data.url,
+                content=request_data.request_body,
+                auth=request_data.auth,
+                params=request_data.query_params
+            )
+        else:
+            response = await self.http_session.request(
```

Comment:
This part of code looks very similar to the lines 25-31. Maybe you can streamline this by using a single call and handling the content type inline

Reply:
> This part of code looks very similar to the lines 25-31. Maybe you can streamline this by using a single call and handling the content type inline

I've done it on purpose. Since this is a async context I prefer to keep it simple